internal/states/statemgr/filesystem.go

```diff
@@ -140,13 +135,45 @@ func (s *Filesystem) WriteState(state *states.State) error {
 }
 
 func (s *Filesystem) writeState(state *states.State, meta *SnapshotMeta) error {
+	s.file = s.file.DeepCopy()
+	if s.file == nil {
+		s.file = NewStateFile()
+	}
+	s.file.State = state.DeepCopy()
+
+	if meta != nil {
+		// Force new metadata
+		s.file.Lineage = meta.Lineage
+		s.file.Serial = meta.Serial
+		log.Printf("[TRACE] statemgr.Filesystem: forcing lineage %q serial %d for migration/import", s.file.Lineage, s.file.Serial)
+	}
+
+	return nil
+}
+
+// PersistState writes state to a tfstate file.
+func (s *Filesystem) PersistState(schemas *tofu.Schemas) error {
+	defer s.mutex()()
+
+	return s.persistState(schemas)
+}
+
+func (s *Filesystem) persistState(schemas *tofu.Schemas) error {
+	// TODO: this should use a more robust method of writing state, by first
```

Comment:
Is there a reason not to do this as part of this change?

Reply:
Oh, sorry, I missed the fact that this is simply copied from the previous implementation. Nevermind!